### PR TITLE
Various bug fixes, including checking of part number format in PartFo…

### PR DESCRIPTION
**Pull request by robertseliger**
…rm; also changed default number variation from 01 to 00. More detail:
1.	For visual consistency, added single quotes around the name of column headings in the HTML text that describes how to upload a BOM.
2.	Restructured the functions Part.parse_part_number and Part.parse_partial_part_number to allow number item variation to be alphanumeric (it was looking for just digits).
3.	Fixed the last remaining name__iexact so that it has a default value.
4.	Added ‘mm’ as unit selection choices in a bunch of places (including DISTANCE_UNITS). Somehow managed to miss millimeters when I first added the units.
5.	Expanded the set of unit selection choices for VALUE_UNITS (e.g., add inductance units).
6.	Fixed two places in forms.py where warnings were not properly being generated.
7.	Changed the default initial part number item variation from ‘01’ to ’00’.
8.	Added part number format verification checks in PartForm.

### CODE REVIEW

File: bom/forms.py [hunk 1]
```diff
@@ -492,8 +492,8 @@ def is_valid_choice(choice, choices):
                         if is_valid_choice(part_data[k], v):
                             setattr(part_revision, k, part_data[k])
                         else:
-                            self.add_warning(None, "'{0}' is an invalid choice of value for '{1}' for part in row {2} . Uploading of this property skipped. "
-                                                   "Part will still be uploaded".format(part_data[k], k, row_count))
+                            self.self.warnings.append("'{0}' is an invalid choice of value for '{1}' for part in row {2} . Uploading of this property skipped. "
```
Comment (mpkasp) [inline bom/forms.py]: Looks like you have a typo here - should only be one `self.`

**Comment by mpkasp** [comment]
@robertseliger Left a comment in-line, but also in tests.TestForms, test_part_form is now failing. Can you look into why?

**Comment by robertseliger** [comment]
Fixed.

 
Rob

**Comment by mpkasp** [comment]
Merging this pull request closes issue #32

**Pull request merged**